test/src/main/java/org/apache/accumulo/test/fate/MultipleStoresIT.java

```diff
@@ -0,0 +1,571 @@
+        new Fate<>(testEnv1, store1, Object::toString, DefaultConfiguration.getInstance());
+    fate1.startDeadReservationCleaner();
+    Fate<SleepingTestEnv> fate2 =
+        new Fate<>(testEnv2, store2, Object::toString, DefaultConfiguration.getInstance());
+    fate2.startDeadReservationCleaner();
+
+    for (int i = 0; i < numFateIds; i++) {
+      FateId fateId;
+      // Start half the txns using fate1, and the other half using fate2
+      if (i % 2 == 0) {
+        fateId = fate1.startTransaction();
+        fate1.seedTransaction("op" + i, fateId, new SleepingTestRepo(), true, "test");
+      } else {
+        fateId = fate2.startTransaction();
+        fate2.seedTransaction("op" + i, fateId, new SleepingTestRepo(), true, "test");
+      }
+      allIds.add(fateId);
+    }
+    assertEquals(numFateIds, allIds.size());
+
+    // Should be able to wait for completion on any fate instance
+    for (FateId fateId : allIds) {
+      fate2.waitForCompletion(fateId);
+    }
+    // Ensure that all txns have been executed and have only been executed once
+    assertTrue(Collections.disjoint(testEnv1.executedOps, testEnv2.executedOps));
+    assertEquals(allIds, Sets.union(testEnv1.executedOps, testEnv2.executedOps));
+
+    fate1.shutdown(1, TimeUnit.MINUTES);
+    fate2.shutdown(1, TimeUnit.MINUTES);
+  }
+
+  @Test
+  public void testDeadReservationsCleanup() throws Exception {
+    testDeadReservationsCleanup(FateInstanceType.META);
+    testDeadReservationsCleanup(FateInstanceType.USER);
+  }
+
+  private void testDeadReservationsCleanup(FateInstanceType storeType) throws Exception {
+    // Tests reserving some transactions, then simulating that the Manager died by creating
+    // a new Fate instance and store with a new LockID. The transactions which were
+    // reserved using the old LockID should be cleaned up by Fate's DeadReservationCleaner,
+    // then picked up by the new Fate/store.
+
+    final String tableName = getUniqueNames(1)[0];
+    // One transaction for each FATE worker thread
+    final int numFateIds =
+        Integer.parseInt(Property.MANAGER_FATE_THREADPOOL_SIZE.getDefaultValue());
+    final boolean isUserStore = storeType.equals(FateInstanceType.USER);
+    final Set<FateId> allIds = new HashSet<>();
+    final FateStore<LatchTestEnv> mockedStore1, store2;
+    final LatchTestEnv testEnv1 = new LatchTestEnv();
+    final LatchTestEnv testEnv2 = new LatchTestEnv();
+    final ZooUtil.LockID lock1 = new ZooUtil.LockID("/locks", "L1", 50);
+    final ZooUtil.LockID lock2 = new ZooUtil.LockID("/locks", "L2", 52);
+    Map<FateId,FateStore.FateReservation> reservations;
+
+    if (isUserStore) {
+      createFateTable(client, tableName);
+      mockedStore1 = EasyMock.createMockBuilder(UserFateStore.class)
+          .withConstructor(ClientContext.class, String.class, ZooUtil.LockID.class)
+          .withArgs(client, tableName, lock1).addMockedMethod("isDeadReservation").createMock();
```

Comment:
For now, I went with your next suggestion with passing the predicate to the stores. This is not as end-to-end because the lock is not actually in ZK but may be good enough for now. Can be done in follow on issue if desired.